Vlbi functionality

MukundiRams opened the pull request:
We have changed how we initialize the hour angle and compute the range of hour angles. Our initial method led to a large discrepancy between our UV coverage that of the old SIMMS. With out current approach, using Astropy to calculate the Greenwich Hour Angle, the two UV coverages almost match.

SpheMakh reviewed (dismissed):
It looks like there is a mismatch between the schema and `runit()` function

```python
──────────────────────── detailed error report follows ─────────────────────────
        :warning: pre-validation of recipe 'recipe' failed                      
        └── step 'addnoise' failed prevalidation                                
            └── unknown parameter 'recipe.addnoise.addnoise'  
```

MukundiRams commented:
> It looks like there is a mismatch between the schema and `runit()` function
> 
> ```python
> ──────────────────────── detailed error report follows ─────────────────────────
>         :warning: pre-validation of recipe 'recipe' failed                      
>         └── step 'addnoise' failed prevalidation                                
>             └── unknown parameter 'recipe.addnoise.addnoise'  
> ```

It seems we removed the addnoise parameter. Instead of allowing the user to give us an SEFD and specifying if they want noise to be added, now we automatically add noise if the SEFD is provided, hence we removed this parameter. I had removed it on all the files except the observe.yaml one, which might be the cause of the mismatch. Hopefully removing it fixes it.

Allycan commented on simms/schemas/schema_observation.yaml:
```diff
@@ -25,7 +25,7 @@ Observation:
       - -30d0m0s
   start_time:
-    info: Observation UTC start time. Use this format DD/MM/YYYY HH:MM:SS
+    info: Observation UTC start time. Use this format YYYY/MM/DDTHH:MM:SS
```
I believe the time should be parsed with dashes (-) instead of forward slashes (/) to follow the ISO time format.

MukundiRams commented on simms/parser_config/simms-cabs.yaml:
```diff
@@ -108,10 +109,14 @@ cabs:
       starttime: 
-        info: Observation start time. For example, 'UTC,05/02/2024,06:15:10' 
-        metavar: '[TIME ZONE,][DD/MM/YYYY,],HH:MM:SS'
+        info: Observation start time in UTC. For example, '2024/03/14T06:15:10' 
```
It is still failing the CI test even after updating the time format.

Allycan commented:
@MukundiRams, it’s still showing time with forward slashes, only T is
updated.

'2024/03/14T06:15:10'



-A


On Wed, 14 May 2025 at 01:19, Mukundi Ramanyimi ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In simms/parser_config/simms-cabs.yaml
> <https://github.com/wits-cfa/simms/pull/81#discussion_r2087759229>:
>
> > @@ -108,10 +109,14 @@ cabs:
>          default: 'J2000,1h0m0s,-31d0m0s'
>          abbreviation: dir
>        starttime:
> -        info: Observation start time. For example, 'UTC,05/02/2024,06:15:10'
> -        metavar: '[TIME ZONE,][DD/MM/YYYY,],HH:MM:SS'
> +        info: Observation start time in UTC. For example, '2024/03/14T06:15:10'
>
> It is still failing the CI test even after updating the time format.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wits-cfa/simms/pull/81#discussion_r2087759229>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANUYATEAKMHIAGJMUCQW6HD26J4XJAVCNFSM6AAAAAB45ZVVM2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQMZYGMZDGMRYGA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

Allycan reviewed (approved):
All looks good.
